### DISCUSSION TITLE
Use connectedhomeip as a Zephyr module

### DISCUSSION THREAD

**Discussion by QuentinCaldeira-eaton** [zephyrproject-rtos/zephyr#77334] [Ideas]
Hi ! I don't know if it's the right place to discuss that, but I think it should be good to have connectedhomeip (Matter) as a Zephyr module. The standalone platform is now working on connectedhomeip side. It needs more test, because I only have an application with Ethernet connectivity. We should at least test Thread, BLE and WiFi, but I don't have a board for that yet.

I don't know what's this imply, but I think it's worth it. Zephyr has the great strength to provide a lot of agnostic features for many boards, and will make it easier for developers for sure. I work with a Nucleo h563zi, but I think my application can be port for any Ethernet Zephyr board with no big deals (except from hal modules).

I could understand that Zephyr doesn't need Matter, and it's true. Also, there is so much more priorities. But if connectedhomeip should become a Zephyr module, I think it's the good time. 

A Feature Request was already made https://github.com/zephyrproject-rtos/zephyr/issues/45993, but I don't think this is actual. Also, if some people want to test the code on others Zephyr ethernet board, I will be glad to send you the app.

**Comment by OgnjenX** [comment]
Hi, I'm also interested in using the Matter protocol in my Zephyr-based firmware. Are you sure it is not currently possible to use CHIP (Matter) as a module with support added to the https://github.com/project-chip/connectedhomeip repo? I would like to help/contribute, but I'm trying to get a better idea of the current state and limitations.
